dFusion/dFusionSpec.md

```diff
@@ -79,10 +79,93 @@ To summarize, here is a list of state that is stored inside the smart contract:
 ## Batch Auction Workflow
 
 The trading workflow consists of the following sequential processes:
+
+0. Account opening, deposits & withdrawals
 1. On-Chain order collection
-2. Finding the batch price: optimization of batch trading surplus (off-chain)
-3. Verifying batch price and trade execution (zkSnark)
-4. Restart with step 1
+2. Transition function from sha to Pedersen hashes (zkSnark)
+3. Finding the batch price: optimization of batch trading surplus (offchain)
+4. Verifying batch price and trade execution (zkSnark)
+5. Processing of pending exits and deposits (zkSnark)
+6. Restart with step 0
+
+### Account opening & token registration
+
+
+#### Registering accounts
+The contract specifies a constant number of accounts, `K`, that can be opened by anyone by providing an `accountIndex` such that `1 <= accountIndex <= K`. 
+The account index is referred to, synonomously, as the account ID.
+
+Upon opening an account, the contract verifies that
+    - there is no account currently occupying the requested index and
+    - the sender does not currently occupy an account slot.
+
+On the level of contract storage, these contraints imply a bijective mapping {1, 2, ..., K} <-> addresses.
+
+Note that: Registering accounts by specified index (rather than incrementing) enables the possiblity for accounts to be closed and account slots to be made available.
+
+#### Registering tokens
+
+Token registration is performed by token address.
+Tokens are indexed incrementally (implying that they can never be removed) from 1 to `T` (the maximum number of allowed tokens).
+Only the contract owner may register tokens by address.
+
+**Note that** the contract does not verify the requested token address is actually that of an ERC20 token. This implies that security is enforced by permissions (i.e. only owner may register tokens).
+
+In the event that a *non-token* is registered as a token, the indexing slot would be permanently occupied, but no successful deposits could ever be made. However, if the registered token does implement a transferFrom successful deposits could occur
```

Comment:
I'd leave this section out. I believe there is a lot that could happen if someone registered a non-token address and a full analysis of the impact may go beyond the scope of this spec. I think the note makes it clear that this is outside our threat model.

Reply:
So exclude just this paragraph or also the one before?